### ISSUE TITLE
Update app colors to align with VADS/USWDS

### ISSUE THREAD

**Issue by narin** [department-of-veterans-affairs/va-mobile-app#7573]
## Description
On December 4, [VADS updated their primary colors to align with USWDS](https://dsva.slack.com/archives/C03R5SBELQM/p1699983323484419). They updated most of the hex codes and [removed primary dark](https://github.com/department-of-veterans-affairs/component-library/pull/974/files#diff-4c8e552054bce5d252f757cbd4130433736912bf2c1b93cfcdc0c94423de92b6).

On January 8, [VADS updated their remaining colors to align with USWDS](https://dsva.slack.com/archives/C01DBGX4P45/p1704385447508359). They updated most of the hex codes and removed 5 additional colors (orange, coolBlue, coolBlueLight, coolBlueLighter, and coolBlueLightest.

The mobile app design system team created a [spreadsheet](https://docs.google.com/spreadsheets/d/146kWOH2rFeO63Y4Dy0MFL8yEn_4M7Cosc-zgbyWB3x8/edit?usp=sharing) for these updates.

## Acceptance Criteria
In the va-mobile-app repo:
- [ ] Reconcile what to do about the 6 colors being removed
- [ ] Determine whether color tokens are ready to be pulled from va-mobile-library repo, or if we will just update the HEX codes in the [this file](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/styles/themes/VAColors.ts)
- [ ] Update existing colors based on the color changes outlined in this [spreadsheet](https://docs.google.com/spreadsheets/d/146kWOH2rFeO63Y4Dy0MFL8yEn_4M7Cosc-zgbyWB3x8/edit?usp=sharing).
- [ ] Notify Brea and Jessica when the updates are complete, so they can update the Figma files.

**Comment by narin** [comment]
@dumathane This ticket replaces #8476 which you asked for some guidance on. Instead of replacing the hex codes directly in [va-mobile-app's VAColors.ts](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/13f663af03144035f2f79460145500777373156f/VAMobile/src/styles/themes/VAColors.ts), we recommend adding the [mobile-tokens](https://www.npmjs.com/package/@department-of-veterans-affairs/mobile-tokens) package as a dependency and referencing the colors we have there (instructions in the README). That way, if VADS updates their colors in the future, you can just update to the latest version of mobile-tokens instead of having to worry about hex codes down the line.

**Comment by theodur** [comment]
I updated the estimate to a 5 because there are a lot of unused colors in the app that need to be parsed through and removed

**Comment by TKDickson** [comment]
OK had a very helpful meeting, here are some notes:
* @brea11y made a quick ["before and after" sandbox](https://www.figma.com/design/BVfWDc8vinwx8kRc1LoqGl/%F0%9F%8F%96-Brea's-Sandbox?node-id=1445-2883&t=Igtw1qkZy9DEOZzJ-0) for visual reference as well. Short version: light mode changes should be imperceptible, by and large, to the naked eye. Dark mode has more changes, but mainly that the grey highlight for many elements (boxes/cards/rows, but not necessarily all of those literal components in the code of course) is a little more grey. 
* There are a number of places right now where we're not displaying the correct colors (ex: things that should be the new grey are split between an almost-black color, and a way-too-light grey). @jessicawoodin volunteered to review to add a specific list, for the things she can see/tell, to make it easier to fix those. Thank you!
* The new alert component isn't in yet, so alert component is just looking not right. I'm going to follow up in Slack about what sequencing makes sense for that.
* I also noticed that basically every component we touched during HSP doesn't have an active state. I'm going to follow up on that in Slack as well (to figure out the colors we want to be using, and the best way to do that work - I'm going to guess it'll end up as a separate ticket).

**Comment by TKDickson** [comment]
Wrote #9302 for the active state colors for HSP components.

**Comment by TKDickson** [comment]
We should still be OK to move forward with this changes, even without the alert component stuff (#8467 and #8468). Right now it looks like the current build is using either light or lightest yellow, and it should actually be using YellowVivid20/#face00, for the sidebar of the yellow alert. ([Design system color reference](https://department-of-veterans-affairs.github.io/va-mobile-library/?path=/docs/design-tokens-colors--docs#yellow))

Will send this back for changes, once the list of additional mappings/fixes is ready.

**Comment by jessicawoodin** [comment]
I pulled the color changes into the [original spreadsheet](https://docs.google.com/spreadsheets/d/146kWOH2rFeO63Y4Dy0MFL8yEn_4M7Cosc-zgbyWB3x8/edit?gid=1956211175#gid=1956211175) and have several questions. I filtered [this tab](https://docs.google.com/spreadsheets/d/146kWOH2rFeO63Y4Dy0MFL8yEn_4M7Cosc-zgbyWB3x8/edit?gid=1956211175#gid=1956211175) to only show the questions in column G. I think the darkmodeGrayWarmDark change will resolve at least one of the issues that @TKDickson found so far. Let me know if you have any other questions for me!

**Comment by theodur** [comment]
@jessicawoodin To answer your questions from the spreadsheet: 
- `primaryTextColor` is currently not used in our dark mode theme, so it doesn't look like we need a dark mode color for that
- I do think the `pickerControls` color should be `uswdsSystemColorGray80` instead, here are the comparisons: Before/after: <details><summary>Screenshot</summary><img src="https://github.com/user-attachments/assets/dec1f960-e7ee-4225-881d-6a0b1200e649" width="49%" />&nbsp;&nbsp;<img src="https://github.com/user-attachments/assets/967a7957-7570-48ff-8e00-a8d713777235" width="49%" /></details>
- It looks like the `menu` color should be `uswdsSystemColorGray80` instead also, here are the comparisons:
Before/after: <details><summary>Screenshot</summary><img src="https://github.com/user-attachments/assets/e6fba66a-6f15-4c28-9ca1-9ab2ba485c58" width="49%" />&nbsp;&nbsp;<img src="https://github.com/user-attachments/assets/8e82631b-397e-4f36-b72c-2e0ef0b6cbae" width="49%" /></details>

- `veteranStatusHome`, `linkRow`, and `announcementBanner` should all definitely be `uswdsSystemColorGray80` IMO. These are all rendered on the home screen and here's the before/after:
Before/after: <details><summary>Screenshot</summary><img src="https://github.com/user-attachments/assets/6619408f-368e-4706-bcec-6b64c5f013cd" width="49%" />&nbsp;&nbsp;<img src="https://github.com/user-attachments/assets/075117ea-49d5-40a5-aa2d-09ca7761354b" width="49%" /></details>

- The `nav` color isn't used in the app in either light or dark mode

- After looking into the snackbar code, the `snackBarText` color isn't used in the app at all despite being in our color file. There's another color called `snackBarTxt` that's actually used for the snackbar text. The light mode color is `vadsColorBaseLighter` and the dark mode color is `vadsColorBaseDarker` which seems correct:
Light/dark: <details><summary>Screenshot</summary><img src="https://github.com/user-attachments/assets/b29afdac-594f-4509-ab14-07e1c936a339" width="49%" />&nbsp;&nbsp;<img src="https://github.com/user-attachments/assets/3f8f0d3c-c9d6-4364-aae9-b50ae927d8f1" width="49%" /></details>
I'm probably going to remove `snackBarText` since it's not used, and rename `snackBarTxt` to just `snackBar` in the app code for consistency.

- `statusDescription` is also no longer used in the app for either light or dark mode
- `closePanel` is also no longer used in either light or dark mode
- `buttonPrimary` and `buttonSecondaryDisabled` are used in the Pagination component for the buttons that navigate forward/back: <details><summary>Screenshot</summary><img src="https://github.com/user-attachments/assets/2f641b0f-fca4-4283-bef1-f5783fb8d41d" width="49%" /></details>

**Comment by jessicawoodin** [comment]
That all makes sense to me! The gray80 looks much better than gray60.

**Comment by TKDickson** [comment]
Much improved! There are a few places that should be greyer rather than blacker in dark mode (not a designer, can you tell?) - the inbox list rows, the prescriptions cards, the message boxes in message details, the disability rating rows, the box in contact VA (and those are the locations I specifically looked at, but the general idea is "all rows, boxes and cards like this in dark mode")

What they look like right now:
<img width="372" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMUZhQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c435b6115679f20f56c00558f588d9854c7bd6d7/image.png" alt="image.png" />
<img width="388" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMUphQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--2e3aea04a301c061cb097e0cbc6d8d37b22e9bfe/image.png" alt="image.png" />
<img width="388" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMU5hQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0c46f617f25e1f904480e9059c4311f5b4f2cd57/image.png" alt="image.png" />
<img width="388" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMVJhQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--1eba6dc8723601df530a7b05e5ff70686055c5aa/image.png" alt="image.png" />



My best guess is that they're currently grey90, but are supposed to be grey80 (based on the before and after comparisons in Brea's sandbox). Screenshots from sandbox:
<img width="281" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWxhQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9557ed8df054ee0a79b360f30250181b4905c95a/image.png" alt="image.png" />
<img width="273" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMXRhQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--dc1493eddcac9a6e71c96fcffb0f8b784d188880/image.png" alt="image.png" />

**Comment by TKDickson** [comment]
Seems like folks are generally on board with my suggestion to do visual QA only (instead of our typical "QA Eng take a look, then UX team does visual QA") on this ticket.

Still TBD who from the UX team would be doing that work, so I'm moving this ticket to blocked for now.

**Comment by TKDickson** [comment]
Reassigned QA to @rbontrager  (part of QA rotation).

A note that the blocker on this is still, afaik, "finding and assigning a UX person to do the VQA (which is the bulk of the testing work on this ticket)". After which point QA can do an extremely quick functional pass if needed, at the discretion of the QA assigned. Every UX person I've talked to about this agreed that plan made sense, and explained to me why they personally were not the right person to do the VQA :). I have no horse in that race, so keeping the UX blocker on.

@ajsarkar28 @drjecker FYI

**Comment by htcollier** [comment]
I spoke with @jessicawoodin about this work today - based on that conversation, figuring out _how_ to even go about doing this VQA feels like it's going to require some time to think through/plan, and needs its own ticket. I'm going to spin up a stub ticket based on what I understand so far.

**Comment by rbontrager** [comment]
@theodur I pulled the latest dev into the PR and now its failing on bundle android for all the detox tests: https://github.com/department-of-veterans-affairs/va-mobile-app/actions/runs/11843289076/job/33003953665.

**Comment by rbontrager** [comment]
Brea approved the ticket in the [PR](https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/9199). The PR ran all the detox tests which passed so functionally this should be good.  Approved by QA.
